src/Command/ExecuteCommand.php

```diff
@@ -44,25 +52,29 @@ protected function configure()
     /**
      * @param \Symfony\Component\Console\Input\InputInterface $input
      * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @throws \InvalidArgumentException
+     * @throws \LogicException
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $io = new SymfonyStyle($input, $output);
 
-        $className = 'Version' . $input->getArgument(static::INPUT_VERSION);
-        $fileName = $className . '.php';
-        $version = $this->migrationsDirectory . '/' . $fileName;
+        $fileName = 'Version' . $input->getArgument(static::INPUT_VERSION) . '.php';
 
-        if (!file_exists($version)) {
-            $io->error(sprintf('Migration: %s does not exist.', $fileName));
+        $finder = new Finder();
+        $finder->files()->in($this->migrationsDirectory)->name($fileName);
+
+        if (!$finder->hasResults()) {
+            $io->error(sprintf('Migration: %s does not exist in %s.', $fileName, $this->migrationsDirectory));
 
             return;
         }
 
-        require_once $version;
-
-        /** @var \BowlOfSoup\CouchbaseMigrationsBundle\Migration\AbstractMigration $migration */
-        $migration = new $className($this->clusterFactory);
+        $migrationFactory = new MigrationFactory($this->clusterFactory);
+        $iterator = $finder->getIterator();
```

Comment:
Add whitelines around the `$iterator` logic for readability.

Reply:
fixed